store/android/src/main/java/com/jakewharton/sdksearch/store/SqlItemStore.kt

```diff
@@ -11,18 +12,21 @@ import javax.inject.Singleton
 internal class SqlItemStore @Inject constructor(
     private val db: BriteDatabase
 ) : ItemStore {
-  private val clearListing by lazy { ItemModel.Clear_listing(db.writableDatabase) }
   private val insertItem by lazy { ItemModel.Insert_item(db.writableDatabase) }
+  private val updateItem by lazy { ItemModel.Update_item(db.writableDatabase) }
   private val queryTermMapper = SqlItem.FACTORY.query_termMapper()
 
-  override suspend fun updateListing(listing: String, items: List<Item>) {
+  override suspend fun updateItems(items: List<Item>) {
     db.inTransaction {
-      clearListing.insert {
-        bind(listing)
-      }
       for (item in items) {
-        insertItem.insert {
-          bind(listing, item.packageName, item.className, item.deprecated, item.link)
+        try {
+          insertItem.insert {
```

Comment:
in general i prefer updates then inserting when the update affects 0 rows just because usually updates happen more often and creating the stack trace as part of the exception is gross but idk how much you care about perf

Reply:
ah, nice. that'd be `OR ABORT` which triggered that?

Reply:
it'd still be `INSERT OR FAIL` but you just do the insert after the update and let it crash if there was a constraint violation.

Reply:
got it. that's much nicer.